Kill the ability to codegen tensor-based broadcasting.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37615 Port existing zero_dim_dispatch optimizations from codegen and remove codegen capability.
* #37614 Kill broadcasting from the codegen layer.
* #37613 Move addmm broadcasting code from codegen layer to native layer.
* #37603 Move addbmm broadcasting from the codegen layer to native layer.
* #37582 Specify _th_ ops in CUDAUnaryOps macros so they are easier to find.
* **#37547 Kill the ability to codegen tensor-based broadcasting.**
* #37546 Move addr broadcasting from codegen layer to native layer.
* #37545 Move broadcasting code for fmod, fmod_ from codegen layer.
* #37544 Move baddbmm broadcasting from codegen layer to native layer.

This shouldn't be used anymore.

Differential Revision: [D21315037](https://our.internmc.facebook.com/intern/diff/D21315037)

gchanan committed:
Kill the ability to codegen tensor-based broadcasting.

This shouldn't be used anymore.

[ghstack-poisoned]

bhosmer commented on aten/src/ATen/function_wrapper.py:
```diff
@@ -887,15 +887,13 @@ def get_broadcast_actuals(broadcast_arg, broadcast_inplace, broadcast_dims):
         # 2) in the broadcast_dims case, this is the broadcasted argument (e.g. "self") followed by the sizes
         #    it is broadcasted to (as an initializer list), so e.g. the specification
         #    "mat1.dim0,mat2.dim1" gets transformed to "self, {mat1.size(0),mat2.size(1)}"
-        if not broadcast_dims:
```
I think case (1) in the comment can be deleted...

gchanan committed:
Update on "Kill the ability to codegen tensor-based broadcasting."


This shouldn't be used anymore.

Differential Revision: [D21315037](https://our.internmc.facebook.com/intern/diff/D21315037)

[ghstack-poisoned]

gchanan committed:
Update on "Kill the ability to codegen tensor-based broadcasting."


This shouldn't be used anymore.

Differential Revision: [D21315037](https://our.internmc.facebook.com/intern/diff/D21315037)

[ghstack-poisoned]

gchanan committed:
Update on "Kill the ability to codegen tensor-based broadcasting."


This shouldn't be used anymore.

Differential Revision: [D21315037](https://our.internmc.facebook.com/intern/diff/D21315037)

[ghstack-poisoned]
